### PR TITLE
CLOUDP-59267:  upgrade all agents in Org

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [CLOUDP-59267](https://jira.mongodb.org/browse/CLOUDP-59267)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] I have tested my code using my local ops manager

## Further comments

```
./bin/mongocli om agents upgrade -P ops2
{
	"automationAgentVersion": "10.14.0.6306-1",
	"biConnectorVersion": "2.13.4"
}
```

### CODE REVIEW

File: internal/cli/atlas_dbusers_list.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func (opts *atlasDBUsersListOpts) initStore() error {
 
 func (opts *atlasDBUsersListOpts) Run() error {
 	listOpts := opts.newListOptions()
-	result, err := opts.store.DatabaseUsers(opts.projectID, listOpts)
+	result, err := opts.store.DatabaseUsers(opts.ProjectID(), listOpts)
```
Comment (gssbzn) [inline internal/cli/atlas_dbusers_list.go:39]: 👍

File: internal/cli/ops_manager_agents_upgrade.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	return json.PrettyPrint(r)
+}
+
+// mongocli ops-manager agents upgrade [--projectId projectId]
+func OpsManagerAgentsUpgradeBuilder() *cobra.Command {
+	opts := &opsManagerAgentsUpgradeOpts{}
+	cmd := &cobra.Command{
+		Use:     "upgrade",
+		Aliases: []string{"upgrade"},
```
Comment (gssbzn) [inline internal/cli/ops_manager_agents_upgrade.go:51]: I think we can remove this

File: internal/description/description.go [hunk 3]
```diff
@@ -87,6 +87,7 @@ const (
 	ListProjects           = "List projects."
 	DownloadLogs           = "Download logs from a log collection job."
 	OpsManager             = "Ops Manager operations."
+	Agents                 = "Manage Ops Manager Agents."
```
Comment (gssbzn) [inline internal/description/description.go:90]: [nit] agents is not a proper name
```suggestion
	Agents                 = "Manage Ops Manager agents."
```

File: internal/store/agents.go [hunk 4]
```diff
@@ -35,3 +39,14 @@ func (s *Store) Agents(projectID, agentType string) (*om.Agents, error) {
 		return nil, fmt.Errorf("unsupported service: %s", s.service)
 	}
 }
+
+// Agents encapsulate the logic to manage different cloud providers
```
Comment (gssbzn) [inline internal/store/agents.go:43]:
```suggestion
// UpgradeAgent encapsulate the logic to manage different cloud providers
```

File: internal/cli/ops_manager_agents_upgrade_test.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"testing"
+
+	om "github.com/mongodb/go-client-mongodb-ops-manager/opsmngr"
+
```
Comment (gssbzn) [inline internal/cli/ops_manager_agents_upgrade_test.go:21]: let's remove the extra space here please

File: internal/cli/ops_manager_agents_upgrade.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/cli/ops_manager_agents_upgrade.go:51]: I would rather change it as
```suggestion
		Aliases: []string{"upgrades"},
```

Comment (gssbzn) [reply internal/cli/ops_manager_agents_upgrade.go]: the thing with that is that `mongocli om agent upgrades` breaks our usage of things like update/list/create where we don't offer updates/lists/creates, also is not fully natural in terms of grammar

File: internal/cli/ops_manager_agents_upgrade.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+	return json.PrettyPrint(r)
+}
+
+// mongocli ops-manager agents upgrade [--projectId projectId]
+func OpsManagerAgentsUpgradeBuilder() *cobra.Command {
+	opts := &opsManagerAgentsUpgradeOpts{}
+	cmd := &cobra.Command{
+		Use:     "upgrade",
+		Aliases: []string{"upgrades"},
```
Comment (gssbzn) [inline internal/cli/ops_manager_agents_upgrade.go:51]: I already answer on the original but adding this alias here breaks our patter of other commands like update/create/list where we don't offer this type of alias

**Pull request merged**